discovery_engine_core/core/src/storage/migrations/20220628114905_feed.sql

```diff
@@ -0,0 +1,54 @@
+--  Copyright 2022 Xayn AG
+--
+--  This program is free software: you can redistribute it and/or modify
+--  it under the terms of the GNU Affero General Public License as
+--  published by the Free Software Foundation, version 3.
+--
+--  This program is distributed in the hope that it will be useful,
+--  but WITHOUT ANY WARRANTY; without even the implied warranty of
+--  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--  GNU Affero General Public License for more details.
+--
+--  You should have received a copy of the GNU Affero General Public License
+--  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+CREATE TABLE IF NOT EXISTS FeedDocument (
+    document BLOB NOT NULL
+        PRIMARY KEY
+        REFERENCES HistoricDocument(document) ON DELETE CASCADE
+);
+
+CREATE TRIGGER IF NOT EXISTS feed_history_sync
+    BEFORE INSERT
+    ON FeedDocument
+    BEGIN
+      INSERT INTO HistoricDocument(document)
+        VALUES (new.document)
+        ON CONFLICT DO NOTHING;
+    END;
```

Comment:
Will this much code to do this?
I would prefer to have all the database logic in one 'place' and not split between rust and sql

Reply:
I would prefer triggers for this use case tbh. it's faster and makes things more reliable.

Especially if we e.g. move documents from the stack to the feed.

Reply:
I'm not strictly against no-triggers, it's just having triggers for this specific kind of use-case is quite nice as it means you don't have to handle the history explicitly.

If we enable foreign constraint checking in sqlite history being there is also enforce through the fk-constraint on the feeds document id so it's not too important. But if we don't enable fk-constraint checking I would strongly recommend it.

Reply:
From my point of view, I don't have a strong preference in either way

Reply:
I'm not strictly against no-triggers, it's just having triggers for this specific kind of use-case is quite nice as it means you don't have to handle the history explicitly.

If we enable foreign constraint checking in sqlite history being there is also enforce through the fk-constraint on the feeds document id so it's not too important. But if we don't enable fk-constraint checking I would strongly recommend it. 

EDIT: I wrote this already yesterday and send it but now it's unsend again???

Reply:
then lets just keep the trigger removed, makes it easier for anyone not familiar with sql(ite)/triggers, I guess